lib/file.js

```diff
@@ -147,6 +150,62 @@ class File extends EventEmitter {
     render.render(this, elem, opts, cb)
   }
 
+  _serve (req) {
+    const res = {
+      status: 200,
+      headers: {
+        // Support range-requests
+        'Accept-Ranges': 'bytes',
+        'Content-Type': this._getMimeType()
+      }
+    }
+    // force the browser to download the file if there is a specific header present
+    if (req.destination !== 'iframe' && req.headers['upgrade-insecure-requests'] === '1') {
+      res.headers['Content-Type'] = 'application/octet-stream'
+      res.headers['Content-Disposition'] = 'attachment'
+    }
```

Comment:
I don't understand this part.   Using a service worker, all the incoming requests should be for a secure page (localhost is a secure origin in chrome while developing).
And if they did want to pipe this into an iframe without downloading the file, I think they should be able to.

Reply:
Yeah, I've already changed this, the upgrade-insecure-requests header was a hack really, to figure out of the user is requesting from a new tab, I've changed this to use destination to exclude iframes, but then realised there's a `document` destination I could just target instead, now you can display the content in an iframe, and open it in a new tab to download it